config/confighttp/confighttp.go

```diff
@@ -0,0 +1,62 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package confighttp defines the HTTP configuration settings.
+package confighttp
+
+import (
+	"net/http"
+
+	"github.com/rs/cors"
+
+	"go.opentelemetry.io/collector/config/configtls"
+)
+
+// HTTPServerSettings defines common settings for a HTTP server configuration.
+type HTTPServerSettings struct {
+
+	// Addr the address for the server to listen on.
+	Addr string `mapstructure:"addr"`
```

Comment:
Should be called `endpoint`. `ReceiverSettings` has endpoint in it but we can't really embed here because this should usable across different types. I think we probably need to use it with at least 1 consumer to flesh out the API. Are there any candidates?

Reply:
It probably makes sense to work towards moving `endpoint` out of `ReceiverSettings`, given that its not relevant for every receiver, e.g. Prometheus, Host Metrics.

In terms of candidates for using this config, I think it would be good to think about how the OTLP receiver should look given that it has both HTTP & gRCP server settings. Also see the Jaeger receiver that supports four Server protocols.